executor/join_test.go

```diff
@@ -215,6 +215,24 @@ func (s *testSuite) TestJoin(c *C) {
 	) `)
 	result.Check(testkit.Rows("<nil>"))
 
+	// test virtual rows are included (issue#5771)
+	result = tk.MustQuery(`SELECT 1 FROM (SELECT 1) t1, (SELECT 1) t2`)
+	result.Check(testkit.Rows("1"))
+	result = tk.MustQuery(`
+		SELECT @NUM := @NUM + 1 as NUM FROM
+		( SELECT 1 UNION ALL
+			SELECT 2 UNION ALL
+			SELECT 3
+		) a
+		INNER JOIN
+		( SELECT 1 UNION ALL
+			SELECT 2 UNION ALL
+			SELECT 3
+		) b,
+		(SELECT @NUM := 0) d;
+	`)
+	result.Check(testkit.Rows("1", "2", "3", "4", "5", "6", "7", "8", "9"))
```

Comment:
another simpler test cast:
```sql
select 1 from (select 1) t1, (select 1) t2;
``

Reply:
That test is included above at lines 219-220. Perhaps should be separated by a newline to indicate two separate test cases.

Reply:
That test is included above at lines 219-220. Perhaps should be separated by a newline to indicate two separate test cases.